File: leetcode/E_168Excel表列名称/convertToTitle.cpp
#include <iostream>
#include <string>
#include <cstring>


class Solution {
public:
    std::string convertToTitle(int columnNumber) {
        if (columnNumber < 27) return std::string{static_cast<char>('A'+columnNumber-1)};
        int l = (columnNumber - 1) / 26;
        int c = (columnNumber - 1) % 26 + 1;
        return convertToTitle(l) + convertToTitle(c);
    }
};

int main(int argc, char *argv[])
{
    int n = atoi(argv[1]);
    Solution sol;
    std::cout << sol.convertToTitle(n) << std::endl;
    return 0;
}

File: leetcode/E_168Excel表列名称/convertToTitle.py
#!/usr/bin/env python
# coding=utf-8
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        if columnNumber < 27:
            return chr(ord('A') + columnNumber - 1)
        l = (columnNumber - 1) // 26
        c = (columnNumber - 1) % 26 + 1
        return self.convertToTitle(l) + self.convertToTitle(c)


if __name__ == '__main__':
    import sys 

    sol = Solution()
    print(f'{(n:=int(sys.argv[1]))}->{sol.convertToTitle(n)}')
